modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/postAuthnHandler/ChallengeQuestionPostAuthnHandlerTestCase.java

```diff
@@ -0,0 +1,743 @@
+            Assert.assertTrue(isChallengeQuestionsRequested(response), "Missing challenge questions post " +
+                    "authenticator not invoked as expected");
+            if (isChallengeQuestionsRequested(response)) {
+                String pastrCookie = Utils.getPastreCookie(response);
+                Assert.assertNotNull(pastrCookie, "pastr cookie not found in response.");
+                EntityUtils.consume(response.getEntity());
+                response = Utils.sendPOSTChallengeQuestionResponse(response, COMMON_AUTH_URL, USER_AGENT, String.format(ACS_URL, config.getApp()
+                        .getArtifact()), httpClient, pastrCookie);
+            }
+            // Check for the missing claim handler post authenticator
+            if (requestMissingClaims(response)) {
+                String pastrCookie = Utils.getPastreCookie(response);
+                EntityUtils.consume(response.getEntity());
+
+                response = Utils.sendPOSTConsentMessage(response, COMMON_AUTH_URL, USER_AGENT, String.format(ACS_URL, config.getApp()
+                        .getArtifact()), httpClient, pastrCookie);
+                EntityUtils.consume(response.getEntity());
+            }
+            String redirectUrl = Utils.getRedirectUrl(response);
+            if (StringUtils.isNotBlank(redirectUrl)) {
+                EntityUtils.consume(response.getEntity());
+                response = Utils.sendRedirectRequest(response, USER_AGENT, ACS_URL, config.getApp().getArtifact(),
+                        httpClient);
+            }
+            String samlResponse = Utils.extractDataFromResponse(response, CommonConstants.SAML_RESPONSE_PARAM, 5);
+            EntityUtils.consume(response.getEntity());
+            response = sendSAMLMessage(String.format(ACS_URL, config.getApp().getArtifact()), CommonConstants
+                    .SAML_RESPONSE_PARAM, samlResponse, httpClient);
+            resultPage = extractDataFromResponse(response);
+            Assert.assertTrue(resultPage.contains("You are logged in as " + config.getUser().getTenantAwareUsername()),
+                    "SAML SSO Login failed for " + config);
+            EntityUtils.consume(response.getEntity());
+
+        } catch (Exception e) {
+            Assert.fail("Missing Challenge Question post authentication handler failed for " + config, e);
+        }
+    }
+
+    @Test(alwaysRun = true, description = "Testing login when the user already has given challenge questions",
+            groups = "wso2.is",
+            dependsOnMethods = {"testAddSP", "testLoginWithEnabledSetting"})
+    public void testLoginWithChallengeQuestions() {
+        try (CloseableHttpClient httpClient = HttpClientBuilder.create().build()) {
+            HttpResponse response;
+            // Update resident IDP property for forcing challenge questions
+            updateResidentIDPProperty(superTenantResidentIDP, FORCE_ADD_PW_RECOVERY_QUESTION, "true", true);
```

Comment:
If you do any change to a config in the server, you must revert back them at the end of the test. Otherwise, all the tests executed after this test will be broken. With this change, after a user enters the credentials, he will be redirected to a page to set challenge questions, which will not be an expected behavior in other tests.